pymeasure/instruments/nf/li5660.py

```diff
@@ -0,0 +1,445 @@
+        validator=strict_discrete_set,
+        values={"DB": "DB, 'dB'",
+                "Reference": "PCNT, ''",
+                "Sensitivity": "PCFS, ''",
+                },
+        map_values=True
+    )
+
+    reference_voltage = Instrument.control(
+        ":CALC1:MATH:VOLT?", ":CALC1:MATH:VOLT %g",
+        """ A property that allow user to set the voltage reference value for normalize calculation. """,
+        validator=truncated_range,
+        values=[1e-9, 1e+1]
+    )
+
+    offset_enable = Instrument.control(
+        ":CALC1:OFFS:STAT?", ":CALC1:OFFS:STAT %s",
+        """ A property that allow user to enable/disable offset for the primary PSD's X output. """,
+        validator=strict_discrete_set,
+        values={"On": 1, "Off": 0},
+        map_values=True
+    )
+
+    primary_offset = Instrument.control(
+        "CALC1:OFFS?",":CALC1:OFFS:STAT ON; :CALC1:OFFS %g",
+        """ A property that allow user to offset for the primary PSD's X output. """,
+        validator=truncated_range,
+        values=[-105, 105]
+    )
+
+    calculation_method = Instrument.control(
+        ":CALC5:MATH?", ":CALC5:MATH %s",
+        """ A property that allow user to set the calculation method for measurement value to be displayed and output. """,
+        validator=strict_discrete_set,
+        values={"Off": "OFF",
+                "Expand": "EXP",
+                "Normalize": "NORM",
+                "Ratio": "RAT"},
+        map_values=True
+    )
+
+    fetch_data = Instrument.measurement(
+        ":FETC?",
+        """ Queries the most recent measurement data. """
+    )
+
+    #############
+    # Frequency #
+    #############
+    frequency = Instrument.measurement(
+        ":FREQ?",
+        """ Return the fundamental wave, primary frequency. """
+    )
+
+    harmonics_pre_enable = Instrument.control(
+        ":FREQ:HARM?", ":FREQ:HARM %d",
+        """ A property that enable/disable the harmonics measurement. """,
+        validator=strict_discrete_set,
+        values={"On": 1, "Off": 0},
+        map_values=True
+    )
+
+    harmonics_pre_order = Instrument.control(
+        ":FREQ:MULT?", ":FREQ:HARM ON; :FREQ:MULT %d",
+        """ A property that control the harmonic order n for measurement (Primary PSD) """,
+        validator=truncated_range,
+        values=[1, 63]
+    )
+    
+    subharmonic_order = Instrument.control(
+        "FREQ:SMUL?", ":FREQ:HARM ON; :FREQ:MULT %d",
+        """ A property that control the subharmonic order m for measurement (Primary PSD) """,
+        validator=truncated_range,
+        values=[1, 63]
+    )
+
+    secondary_frequency = Instrument.measurement(
+        ":FREQ2?",
+        """ Return the secondary frequency used with detection modes DUAL2 and CASCADE. """
+    )
+
+    harmonics_sec_enable = Instrument.control(
+        ":FREQ2:HARM?",":FREQ2:HARM %d",
+        """ A property that control the harmonic measurement (enabled or disabled) (secondary PSD)""",
```

Comment:
It would be nice to (explicitely) mention the possible values in the docstring. You could also consider using booleans rather than the "On" and "Off" strings.